File: app/src/main/java/com/incendiary/android/di/ConfigPersistent.java
package com.incendiary.android.di;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Scope;

import com.incendiary.android.di.component.ConfigPersistentComponent;

/**
 * A scoping annotation to permit dependencies conform to the life of the
 * {@link ConfigPersistentComponent}
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface ConfigPersistent {
}

File: app/src/main/java/com/incendiary/androidboilerplate/di/ConfigPersistent.kt
package com.incendiary.androidboilerplate.di

import com.incendiary.androidboilerplate.di.component.ConfigPersistentComponent
import javax.inject.Scope

/**
 * A scoping annotation to permit dependencies conform to the life of the
 * [ConfigPersistentComponent]
 */
@Scope
@Retention(AnnotationRetention.RUNTIME)
annotation class ConfigPersistent
